packages/material-ui/src/Tooltip/Tooltip.js

```diff
@@ -115,21 +115,22 @@ function Tooltip(props) {
   const handleRef = useForkRef(children.ref, handleOwnRef);
 
   React.useEffect(() => {
-    if (childNode) {
-      warning(
-        !childNode.disabled ||
-          (childNode.disabled && isControlled) ||
-          (childNode.disabled && title === '') ||
-          childNode.tagName.toLowerCase() !== 'button',
-        [
-          'Material-UI: you are providing a disabled `button` child to the Tooltip component.',
-          'A disabled element does not fire events.',
-          "Tooltip needs to listen to the child element's events to display the title.",
-          '',
-          'Place a `div` container on top of the element.',
-        ].join('\n'),
-      );
-    }
+    warning(
+      !(
```

Comment:
That's is, maybe we can write our own warning method that uses warn internally but reverse the condition logic. It's more intuitive. Maybe we don't need a different if branch, what we need it a direct condition logic, not a reversed one.